.github/workflows/10-linters-tests-image.yaml

```diff
@@ -1,15 +1,23 @@
-# This workflow run linters and tests.
-# It is supposed to be run on each push to main branch, as well as for each push in pull request.
+# This workflow run linters and tests, and publishes a new image if there is a release or
+# push to main. It is supposed to be run on each push to main branch, on each release, as
+# well as for each push in pull request.
 
-name: Linters
+name: Code and Image workflow
 
 on:
   pull_request:
     branches: [ '*' ]
+  push:
+    branches:
+      - "main"
+    tags:
+      - 'v*'
 
 env:
   GO_VERSION: "1.21"
-  IMAGE: linode/linode-cosi-driver
+  REGISTRY: docker.io
+  REPOSITORY: linode
+  IMAGE: linode-cosi-driver
```

Comment:
did you ever track down getting this made?

Reply:
Tested the whole CI setup here:
- https://github.com/shanduur/linode-cosi-driver-test

Artifacts published on my private Docker Hub:
- https://hub.docker.com/r/shanduur/linode-cosi-driver/tags